pkg/trait/trait.go

```diff
@@ -79,11 +79,15 @@ func newEnvironment(ctx context.Context, c client.Client, integration *v1.Integr
 
 	if kit == nil {
 		kit, err = getIntegrationKit(ctx, c, integration)
-		if err != nil {
+		if err != nil && !k8serrors.IsNotFound(err) {
```

Comment:
Could that be a race between the moment the kit is assigned to the Integration and the cached client is refreshed?

It's an invariant of the system that when the Integration is assigned a kit, the kit exists.

Reply:
No. The error was quite subtle. During an upgrade the, kit has to be rebuilt and at that point, the `integration.status.integrationkit` is assigned an [empty/blank integrationkit](https://github.com/apache/camel-k/blob/main/pkg/controller/integration/build_kit.go#L76). However, the tests assume a check for a nil intergationkit on `integration.status` not an empty one. The result is that the error [L83](https://github.com/apache/camel-k/blob/main/pkg/trait/trait.go#L83) is returned, reported in the operator log and the process repeated ad infinitum, even when the integrationkit has been rebuilt and now available to be reassinged to `integration.status.integrationkit`.

What the user sees in an integrationkit fully rebuilt but an integration stuck in the `Building Kit` phase.

Reply:
Can `integration.status.integrationkit` be set to `nil` instead of an empty kit? What is the rational to set it to an empty kit?

Reply:
That I don't know. It was before my time and assumed it was required to be empty rather than nil.

Reply:
Let's try that then, it seems it would fix the issue, without violating the kit invariant. WDYT?